checkov/terraform/graph_builder/variable_rendering/evaluate_terraform.py

```diff
@@ -98,6 +98,42 @@ def replace_string_value(original_str: Any, str_to_replace: str, replaced_value:
     return string_without_interpolation.replace(str_to_replace, str(replaced_value))
 
 
+def _find_new_value_for_interpolation(origin_str: str, str_to_replace: str, new_value: str) -> str:
+    """
+    This function checks whether we should escape the interpolated value, to avoid syntax error.
+    Example:
+        origin_str = "${lookup({'a': ${local.protocol1}},\"a\",\"https\")}"
+        If we don't escape local.protocol1, the lookup function will fail, i.e -
+          invalid - ${lookup({'a': local.protocol1},\"a\",\"https\")}
+          valid - ${lookup({'a': 'local.protocol1'},\"a\",\"https\")}
+    Default to return is new_value
+    """
+    try:
+        # First part - checking if not-escaped is valid.
+        not_escaped = origin_str.replace(str_to_replace, new_value)
+        first_evaluated = evaluate_terraform(not_escaped)
+        if abs(len(not_escaped) - len(first_evaluated)) != 3:
+            # checking if the len difference != 3 checks if we didn't only remove the '${}'
+            return new_value
+        second_evaluated = _try_evaluate(first_evaluated)
+        if first_evaluated != second_evaluated and abs(len(not_escaped) - len(second_evaluated)) != 3:
```

Comment:
Can we put the expression inside a function so its meaning would be clearer?

Reply:
good idea, done